api/src/main/java/org/pmiops/workbench/mandrill/MandrillServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.pmiops.workbench.mandrill;
+
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.json.JSONObject;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.mandrill.ApiException;
+import org.pmiops.workbench.mandrill.api.MandrillApi;
+import org.pmiops.workbench.mandrill.model.MandrillApiKeyAndMessage;
+import org.pmiops.workbench.mandrill.model.MandrillMessage;
+import org.pmiops.workbench.mandrill.model.MandrillMessageStatuses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import javax.inject.Provider;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Service
+public class MandrillServiceImpl implements MandrillService {
+
+  private static final Logger log = Logger.getLogger(MandrillServiceImpl.class.getName());
+  private final Provider<WorkbenchConfig> configProvider;
+  private final Provider<MandrillApi> mandrillApiProvider;
+
+  @Autowired
+  public MandrillServiceImpl(Provider<WorkbenchConfig> configProvider,
+                             Provider<MandrillApi> mandrillApiProvider) {
+    this.configProvider = configProvider;
+    this.mandrillApiProvider = mandrillApiProvider;
+  }
+
+  @Override
+  public MandrillMessageStatuses sendEmail(MandrillMessage email) {
+    MandrillApi mandrillApi = mandrillApiProvider.get();
+    String apiKey = readMandrillApiKey();
+    MandrillApiKeyAndMessage keyAndMessage = new MandrillApiKeyAndMessage();
+    keyAndMessage.setKey(apiKey);
+    keyAndMessage.setMessage(email);
+    try {
+      MandrillMessageStatuses msgStatuses = mandrillApi.send(keyAndMessage);
+      log.log(Level.INFO, "Message Status: " + msgStatuses.toString());
+      return msgStatuses;
+    } catch (ApiException e){
+      log.log(Level.WARNING, "Sending email via Mandrill Failed.");
+      return null;
+    }
+  }
+
+  private String readMandrillApiKey() {
+    String bucketName = configProvider.get().googleCloudStorageService.credentialsBucketName;
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    String content = new String(storage.get(bucketName, "mandrill-keys.json").getContent());
+    JSONObject mandrillKeys = new JSONObject(content);
+    return mandrillKeys.getString("api-key");
```

Comment:
The above lines should all be in a method defined in `CloudStorageService` and not here. You should be calling that service to get the key value.

Reply:
Additionally, let's look at abstracting this a little further. If we expand the concept of `mandrill-keys.json` to be larger, say `mandrill-info.json` (or something better-named) and add kv-pairs for the api key, from address, and perhaps something else that might be needed in the future. In other systems I've worked on, we also needed automatic CC and BCC addresses added that were environment-specific.